app/src/main/res/layout/fragment_login.xml

```diff
@@ -69,6 +77,7 @@
             android:text="@string/log_in"
             android:textColor="@color/black"
             android:textSize="@dimen/text_medium_size"
+            android:onClick="@{() -> loginViewModel.nullValidation()}"
```

Comment:
Also here. It's more preferable to specify the interactions of the UI with Kotlin. XML doesn't provide a good way to track errors or debugging features.